### DISCUSSION TITLE
How to use the trace-ignore-plugin to filter the following cases?

### DISCUSSION THREAD

**Discussion by sevenyby** [apache/skywalking#8702] [Q&A] [answered]
Environment: CentOS 7, ES 7.17, SkyWalking 8.9.1

SkyWalking Plugins: apm-trace-ignore-plugin-8.9.0.jar, apm-customize-enhance-plugin-8.9.0.jar

App: SpringBoot 2.2.6.RELEASE + Swagger 3

--------------------------------------------------------------------------------------------------------------

My original intention is to use SkyWalking to track API calls between multiple services, as well as calls between methods within each interface. So I introduced the trace-ignore-plugin to filter some content that I don't want to be tracked. 

E.g:

![image](https://user-images.githubusercontent.com/9942770/158929741-8aefeb7e-bcc0-4862-b5e0-f94c605dfd8c.png)

Like what's in the picture above, none of them are interfaces that I've defined. I don't want them to be traceable. But I don't know how to filter by the plugin.

E.g 2:

Get:/
null:null

I don't know why an interface like the one above would be called and both fail. I also don't want them to be tracked.

Hope to get your help, thank you very much.

**Comment by wu-sheng** [comment, marked as answer]
> Like what's in the picture above, none of them are interfaces that I've defined. I don't want them to be traceable. But I don't know how to filter by the plugin.

This is how your codes work, not about you defined it or not. Ignore plugin has its document, https://skywalking.apache.org/docs/skywalking-java/v8.9.0/en/setup/service-agent/java-agent/agent-optional-plugins/trace-ignore-plugin/

If you want to dig more details, UTs may help. https://github.com/apache/skywalking-java/blob/main/apm-sniffer/optional-plugins/trace-ignore-plugin/src/test/java/org/apache/skywalking/apm/plugin/trace/ignore/TracePathMatcherTest.java

**Reply by sevenyby** [reply]
Thanks, I will try some more based on them.

**Reply by WilliamShoww** [reply]
I have same question, how do you handler?

**Reply by sevenyby** [reply]
> I have same question, how do you handler?

In fact, I tried many methods and didn't eliminate them. Maybe I haven't mastered the skills yet. So I can only act as if I can't see them.

**Reply by WilliamShoww** [reply]
I kown why not filter config endpoint; because springmvc transform `trace-ignore-plugin`'s operationName is 'SpringAsync', the opearatorName is not endpoint path;
![image](https://user-images.githubusercontent.com/19343209/164194536-5cb4985e-55f5-41c7-8757-bb669472842a.png)
![image](https://user-images.githubusercontent.com/19343209/164194604-883fb365-8a16-4632-825d-ca284a377b02.png)

**Reply by WilliamShoww** [reply]
@wu-sheng  I think it's BUG; should fix it. Or i use less some plugins(please tell me less what plugins)

**Reply by WilliamShoww** [reply]
> 我知道为什么不过滤配置端点；因为 springmvc transform`trace-ignore-plugin`的 operationName 是 'SpringAsync'，operatorName 不是端点路径； ![图片](https://user-images.githubusercontent.com/19343209/164194536-5cb4985e-55f5-41c7-8757-bb669472842a.png) ![图片](https://user-images.githubusercontent.com/19343209/164194604-883fb365-8a16-4632-825d-ca284a377b02.png)

I change the source code, add logger

**Reply by wu-sheng** [reply]
I don't know much about how every plugin works. You could read the source codes and figure out why it works like this.
The ignoring mechanism is also a plugin only, the operation name is set by some plugins.

Mostly, I am just working on design, code review. Plugin bugs are definitely countless, I can't know all even most of it, because generally for my daily work, I am not using Java agent.

You should know, open-source project is being used in many different cases. Those who are using this, need to do some research on your own, once you have a certain conclusion, we could discuss whether this is reasonable or a bug.

In your current description, it just says there is an operation name called SpringAsync, there is nothing special, it means there is a request may process crossing two thread, so the plugin add a `async` span, which begins a new tracing context. That is all

**Reply by WilliamShoww** [reply]
Ok， i research more.
